node-red-node-wot/src/wot-property.js

```diff
@@ -1,174 +1,176 @@
+                                shape: "ring",
+                                text: "Observe error",
+                            })
+                        }
+                    )
+                } catch (err) {
+                    console.warn("[warn] property observe error. try again. error: " + err)
+                    node.status({
+                        fill: "red",
+                        shape: "ring",
+                        text: "Observe error",
+                    })
+                }
+                if (ob) {
+                    node.status({
+                        fill: "green",
+                        shape: "dot",
+                        text: "connected",
+                    })
+                    break
+                }
+                await (() => {
+                    return new Promise((resolve) => {
+                        setTimeout(() => {
+                            resolve()
+                        }, 500)
+                    })
+                })()
+            }
         })
-    })
 
-    node.on('close', function () {
-      if (removed) {
-        // This node has been deleted
-      } else {
-        // This node is being restarted
-      }
-      done()
-    })
-  }
-  RED.nodes.registerType('read-property', readPropertyNode)
+        node.on("input", async (msg, send, done) => {
+            if (!consumedThing) {
+                node.error("[error] consumedThing is not defined.")
+                done("consumedThing is not defined.")
+                return
+            }
+            const uriVariables = config.uriVariables ? JSON.parse(config.uriVariables) : undefined
+            consumedThing
+                .readProperty(config.property, { uriVariables: uriVariables })
+                .then(async (resp) => {
+                    let payload
+                    try {
+                        payload = await resp.value()
+                    } catch (err) {
+                        console.error(`[error] failed to get value. name: ${config.property}, error: `, err.toString())
+                        done(`[error] failed to get value. name: ${config.property}, error: ${err.toString()}`)
+                    }
+                    node.send({ payload, topic: config.topic })
+                    node.status({
+                        fill: "green",
+                        shape: "dot",
+                        text: "connected",
+                    })
+                    done()
+                })
+                .catch((err) => {
+                    node.warn(err)
+                    node.status({
+                        fill: "red",
+                        shape: "ring",
+                        text: "Response error",
+                    })
+                    done(err)
+                })
+        })
 
-  function writePropertyNode(config) {
-    RED.nodes.createNode(this, config)
-    let node = this
+        node.on("close", function () {
```

Comment:
Sorry, I must have missed it.
As with other nodes, the arguments must be `removed` and `done`.
```
        node.on("close", function (removed, done) {
```